kernel/src/snapshot.rs

```diff
@@ -175,6 +182,10 @@ impl Snapshot {
         if let Some(version) = version {
             commit_files.retain(|log_path| log_path.version <= version);
         }
+        // only keep commit files above the checkpoint we found
+        if let Some(checkpoint_file) = checkpoint_files.first() {
```

Comment:
I tripped over this by accident when one of the new unit tests failed because it had read 1 json in addition to 5 checkpoint parts. It's technically just a performance bug, since log replay is perfectly happy to ignore the duplicated actions in the checkpoint after having read them from the json.